### DISCUSSION TITLE
Does a live stream flow through ring servers?

### DISCUSSION THREAD

**Discussion by djm181** [dgreif/ring#1169] [Q&A]
When a live call is established with a ring camera, I know that the call is established via the ring servers, but once the RTP/SIP handshake is all complete, if the client and the camera are on the same WiFi network, are the packets for the video actually streaming out over the internet and back in, or does the RTC connection get established directly between the devices?

**Comment by dgreif** [comment]
It depends 😄 
* If you are using Ring Edge, with a camera directly connected to a Ring Alarm Pro, streaming can happen directly from the Ring Alarm Pro to the stream recipient (server running `ring-client-api` or `homebridge-ring`) if network conditions are good. If the server doesn't have direct network visibility to the Ring Alarm Pro, then streams go through the Ring servers.
* If you are not using Ring Edge, _all_ streams go through the Ring servers
